src/upload/imgurprovider.h

```diff
@@ -0,0 +1,47 @@
+/*
+    <one line to give the program's name and a brief idea of what it does.>
+    Copyright (C) 2013  PCMan <email>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef LXIMAGE_IMGURPROVIDER_H
+#define LXIMAGE_IMGURPROVIDER_H
+
+#include <QNetworkAccessManager>
+
+#include "provider.h"
+
+namespace LxImage {
+
+/**
+ * @brief Create uploads to Imgur's API
+ */
+class ImgurProvider : public Provider
+{
+    Q_OBJECT
+
+public:
+
+    virtual Upload *upload(QIODevice *device);
+
+private:
+
+    QNetworkAccessManager mManager;
```

Comment:
Just curious... I don't know how "heavy" the `QNetworkAccessManager` objects are, but the Qt doc says: `One QNetworkAccessManager should be enough for the whole Qt application.`.... then wouldn't it be feasible to make this object be a static member of the `Provider` class?